crates/symbolicator-service/src/services/symbolication/js.rs

```diff
@@ -402,6 +403,13 @@ fn fixup_webpack_filename(abs_path: &str) -> String {
     }
 }
 
+/// Joins a path to a base URL without normalizing `..` segments.
+fn nonstandard_path_join(base: &Url, path: &str) -> Option<String> {
+    let path = path.replace("..", "__dotdot__");
+    let result = base.join(&path).ok()?.to_string();
+    Some(result.replace("__dotdot__", ".."))
```

Comment:
As we have the same problem with `./` paths that we want to preserve, how about we make this into `__dotslash__`, to cover both patterns, as `/../XX` will just be turned into `/.__dotslash__XX` and then be turned back as well.

Also, should we use our custom `join_url` for this? Not sure if we will ever get filenames starting with `~/` here?

I would even argue that we can also special-case `webpack://` here?

Webpack can output some weird-ass `source` filenames, such as `webpack:///../node_modules/prismjs/components/ ../node_modules/prismjs/components/ lazy ^\\.\\/prism\\-.*\\.min$ namespace object`